core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -121,6 +120,24 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Test
       None
     )
     val workspaceService = mock[WorkspaceService](RETURNS_SMART_NULLS)
+    val billingProject = RawlsBillingProject(RawlsBillingProjectName("test-azure-bp"),
+                                             CreationStatuses.Ready,
+                                             None,
+                                             None,
+                                             billingProfileId = Some(UUID.randomUUID().toString)
+    )
+    when(workspaceService.withBillingProjectContext(any(), any())(any())).thenAnswer { invocation =>
+      (invocation.getArgument(2): RawlsBillingProject => Future[Workspace])(billingProject)
+    }
+    when(bpDAO.getBillingProfile(any[UUID], any[RawlsRequestContext])).thenReturn(
```

Comment:
Are these changes necessary because before the hardcoded project was being returned?

Reply:
Correct.